woodwork/utils.py

```diff
@@ -442,3 +442,15 @@ def _update_progress(start_time, current_time, progress_increment,
         callback_function(progress_increment, new_progress, total, unit, elapsed_time)
 
         return new_progress
+
+
+def _infer_datetime_format(dates, n=100):
+    """Helper function to infer the datetime format of the first n non-null rows of a series
+    Args:
+        dates (Series): Series of string or datetime string to guess the format of
+        n (int): the maximum number of nonnull rows to sample from the series
+    """
+    first_n = dates.dropna().head(n)
```

Comment:
Should we take the first n or sample the data randomly? If the data happens to be sorted by date, we might not have a good representation of all the dates present which could lead to incorrect inference.

Reply:
randomly sampling might seem like a better approach, but talking to Max, we should do the first N rows allows for consistency. A dataset tested multiple times returns the inferred datetime format. We can solve this in the future with a random seed, although for Dask we might always want to do first N rows.

Reply:
Fair point. 

As a follow up, what happens if we infer incorrectly based on the first n? Does our initialization break completely and we require users to specify the format? If it does break, is the error meaningful? (I'm also not sure if we can have the same issue with the code in main, FWIW).

Reply:
@thehomebrewnerd for the question `if we infer incorrectly`, Would this be if `pd.to_datetime` was passed the incorrect format, and raised an error? WW code says that it would raise an error
https://github.com/alteryx/woodwork/blob/278aa96fbe78f0e8ed406dcf2d290bb0c015b4b3/woodwork/type_sys/utils.py#L28
- I think either it raises an error, and asks the user to specify OR we coerce the incorrect values to pd.NAT (while warning the user).

Reply:
@gsheni The code you reference is in the Datetime inference function, but we won't be using a format string then unless the user has specified it. We don't get the inferred format string that gets saved on the logical type until after our type inference has run and we attempt to convert the column to a `datetime64` dtype by calling the transform function.

Is is possible that our inference function infers that the column is a datetime, and then fails during dtype conversion if the format returned from the first n rows doesn't match the format that was identified by pandas during the first inference function call?

I'll just have to dig into this more to understand. I think it might be odd behavior if our inference correctly identified a column as datetime and then type conversion failed because we incorrectly specified the wrong format by looking at only the first n rows to get the format string. I'm not sure if that's possible though. I'll test it out...